lib/list-all-files-github.ts

```diff
@@ -1,22 +1,23 @@
 
 import axios from "axios"
-import { ALL_DOCS_PATH } from "../../const"
-import { DocPath, ListDocsResponse } from "./docs.type"
-import { headerGithub } from "../../utils/utils"
+import { headerGithub } from "../utils/utils"
+import { FilePathAndUrl, ListGithubFilesResponse } from "./list-all-files-github-type"
 
-
-export async function ListAllFileInDocs(): Promise<DocPath[]> {
+// ListAllFileInDocs: will return a all of url (navigate to get encrypt content) and path (where is the file in the repo ex./pages/...)
+// includeString is optional we will use it with path like path.includes(includeString)
+// by default it will only include .mdx file
+export async function ListAllFilesInGithub(githubRepoLink:string, includeString:string = ".mdx"): Promise<FilePathAndUrl[]> {
 
     try {
-        let allUrlPath:DocPath[] = []
+        let allUrlPath:FilePathAndUrl[] = []
 
         const headers = headerGithub()
         if (headers === null){
```

Comment:
Are there any no value cases like `undefined` or `blank string`? just checking cause I'm not sure what headerGithub() can return.

Reply:
`underfined` is not possible 
if github key not provide it will return a null

![image](https://github.com/user-attachments/assets/d03b58fa-ab59-4bb9-bf91-4fceb05b82b9)